code/model/DMSDocumentSet.php

```diff
@@ -64,70 +80,69 @@ public function getCMSFields()
+                // Document listing
+                $gridFieldConfig = GridFieldConfig::create()
+                    ->addComponents(
+                        new GridFieldToolbarHeader(),
+                        new GridFieldFilterHeader(),
+                        new GridFieldSortableHeader(),
+                        new GridFieldDataColumns(),
+                        new GridFieldEditButton(),
+                        // Special delete dialog to handle custom behaviour of unlinking and deleting
+                        new DMSGridFieldDeleteAction(),
+                        new GridFieldDetailForm()
+                    );
 
-            if (class_exists('GridFieldSortableRows')) {
-                $sortableComponent = new GridFieldSortableRows('DocumentSort');
-                // setUsePagenation method removed from newer version of SortableGridField.
-                if (method_exists($sortableComponent, 'setUsePagination')) {
-                    $sortableComponent->setUsePagination(false)->setForceRedraw(true);
+                if (class_exists('GridFieldPaginatorWithShowAll')) {
+                    $paginatorComponent = new GridFieldPaginatorWithShowAll(15);
+                } else {
+                    $paginatorComponent = new GridFieldPaginator(15);
                 }
-                $gridFieldConfig->addComponent($sortableComponent);
-            }
+                $gridFieldConfig->addComponent($paginatorComponent);
 
-            // HACK: Create a singleton of DMSDocument to ensure extensions are applied before we try to get display fields.
-            singleton('DMSDocument');
-            $gridFieldConfig->getComponentByType('GridFieldDataColumns')
-                ->setDisplayFields(Config::inst()->get('DMSDocument', 'display_fields'))
-                ->setFieldCasting(array('LastChanged' => 'Datetime->Ago'))
-                ->setFieldFormatting(
-                    array(
-                        'FilenameWithoutID' => '<a target=\'_blank\' class=\'file-url\' href=\'$Link\'>$FilenameWithoutID</a>'
-                    )
-                );
+                if (class_exists('GridFieldSortableRows')) {
+                    $sortableComponent = new GridFieldSortableRows('DocumentSort');
+                    // setUsePagination method removed from newer version of SortableGridField.
+                    if (method_exists($sortableComponent, 'setUsePagination')) {
+                        $sortableComponent->setUsePagination(false)->setForceRedraw(true);
+                    }
+                    $gridFieldConfig->addComponent($sortableComponent);
+                }
 
-            // Override delete functionality with this class
-            $gridFieldConfig->getComponentByType('GridFieldDetailForm')
-                ->setItemRequestClass('DMSGridFieldDetailForm_ItemRequest');
+                // HACK: Create a singleton of DMSDocument to ensure extensions are applied before we try to get display fields.
```

Comment:
Do you think it might be worth trying to remove this "hack" now?

Theoretically you could do the same thing like this:

```php
$displayFields = DMSDocument::create()->config()->get('display_fields');
```